core/vm/eof_control_flow.go

```diff
@@ -83,8 +77,10 @@ func validateControlFlow(code []byte, section int, metadata []*functionMetadata,
 		case JUMPF:
 			arg, _ := parseUint16(code[pos+1:])
 			newSection := metadata[arg]
-			if have, limit := currentStackMax+int(newSection.maxStackHeight)-int(newSection.inputs), int(params.StackLimit); have > limit {
-				return 0, fmt.Errorf("%w: at pos %d", ErrStackOverflow{stackLen: have, limit: limit}, pos)
+			haveLimit := currentStackMax + newSection.maxOuterHeight()
+			wantLimit := int(params.StackLimit)
```

Comment:
I think it's too much like this. It's fine to compare with `int(params.StackLimit)` directly.
BTW the only reason why we need a conversion here is because `StackLimit` is a typed constant. If we removed the type, it would be possible to compare it against any other integer type directly. Might be a useful change to make.

Reply:
i.e.

```
package params

const StackLimit = 1024
```

instead of

```
const StackLimit uint64 = 1024
```

Reply:
For sure, that can easily be done. However, I pushed a different type of change, as an example: https://github.com/ethereum/go-ethereum/pull/30418/commits/363f4ad1edfdb45e34f02ee6aef71cec35b944c2 . If you think this is a good approach, then I can apply the same approach in some more places. Otherwise I'll revert that commit. 


WDYT?

Reply:
It looks a lot better now!